catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorFunctionListTest.java

```diff
@@ -0,0 +1,152 @@
+				{R.string.formula_editor_function_mod, R.string.formula_editor_function_mod_parameter, "mod"},
+				{R.string.formula_editor_function_arcsin, R.string.formula_editor_function_arcsin_parameter, "arcsin"},
+				{R.string.formula_editor_function_arccos, R.string.formula_editor_function_arccos_parameter, "arccos"},
+				{R.string.formula_editor_function_exp, R.string.formula_editor_function_exp_parameter, "exp"},
+				{R.string.formula_editor_function_arctan, R.string.formula_editor_function_arctan_parameter, "arctan"},
+				{R.string.formula_editor_function_floor, R.string.formula_editor_function_floor_parameter, "floor"},
+				{R.string.formula_editor_function_ceil, R.string.formula_editor_function_ceil_parameter, "ceil"},
+				{R.string.formula_editor_function_max, R.string.formula_editor_function_max_parameter, "max"},
+				{R.string.formula_editor_function_min, R.string.formula_editor_function_min_parameter, "min"},
+				{R.string.formula_editor_function_length, R.string.formula_editor_function_length_parameter, "length"},
+				{R.string.formula_editor_function_letter, R.string.formula_editor_function_letter_parameter,
+						"letter"},
+				{R.string.formula_editor_function_join, R.string.formula_editor_function_join_parameter, "join"},
+				{R.string.formula_editor_function_number_of_items, R.string
+						.formula_editor_function_number_of_items_parameter, "number of items"},
+				{R.string.formula_editor_function_list_item, R.string.formula_editor_function_list_item_parameter,
+						"list item"},
+				{R.string.formula_editor_function_contains, R.string.formula_editor_function_contains_parameter,
+						"contains"}
+		});
+	}
+
+	@Parameter
+	public @StringRes int formulaEditorFunction;
+
+	@Parameter(1)
+	public @StringRes int formulaEditorFunctionParameter;
+
+	@Parameter(2)
+	public String testName;
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("FormulaEditorFunctionListTest");
+		script.addBrick(new ChangeSizeByNBrick(0));
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testFunctionsListElements() {
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_change_size_by);
+		onBrickAtPosition(1).onChildView(withId(R.id.brick_change_size_by_edit_text)).perform(click());
+
+		onView(withText(R.string.formula_editor_functions)).perform(click());
+
+		String formulaEditorFunctionString = UiTestUtils.getResourcesString(formulaEditorFunction);
+		String formulaEditorFunctionParameterString = UiTestUtils.getResourcesString(formulaEditorFunctionParameter);
+
+		String editorFunction = formulaEditorFunctionString + formulaEditorFunctionParameterString;
+		String selectedFunctionString = getSelectedFunctionString(editorFunction);
+
+		onData(allOf(is(instanceOf(String.class)), is(editorFunction)))
```

Comment:
if im not mistaken, we talked about this today. so is this also fixable / possible to do in one call now via just clicking on the childView with the string instead of the id?

Reply:
But why do you want to click on the childView with the Text? 
The view element is selected with a string already:
`onData(allOf(is(instanceOf(String.class)), is(editorFunction)))`

In my opinion 	using `.onChildView(withId(R.id.fragment_formula_editor_list_item))` is more elegant than using withText(editorFunction)

Nevertheless i still did some optimizations on this test.

Reply:
And yes, using `.onChildView(withText(editorFunction))` also works